### PR TITLE
[RPW]Adding docs for `ICoreRenderPipelinePreferenceProvider`

### PR DESCRIPTION
**Pull request by alex-vazquez**
### Purpose of this PR
Small docs about the `ICoreRenderPipelinePreferencesProvider`.

---
### Testing status
Check the docs are correctly displayed in a markdown viewer

### CODE REVIEW

File: com.unity.render-pipelines.core/Documentation~/exposing-preferences.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        class Styles
+        {
+            public static readonly GUIContent myBoolLabel = EditorGUIUtility.TrTextContent("MyBool", "This bool toggles my feature");
+        }
+
+        public List<string> keywords => new List<string>() {Styles.myBoolLabel.text};
+        public GUIContent header => EditorGUIUtility.TrTextContent("MyPreferenceSection", "Group of my preferences");
+
+        public static bool s_MyBoolPreference;
```
Comment (RSlysz) [inline com.unity.render-pipelines.core/Documentation~/exposing-preferences.md:18]: Should you add info why static value are better here?

**Review by RSlysz — COMMENTED** [review]
I let documentation do a pass. They are more expert than me here :)

File: com.unity.render-pipelines.core/Documentation~/exposing-preferences.md (hunk 1 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Documentation~/exposing-preferences.md:18]: It was simple example of the usage, I think a better approach should be a preferencekey or a ScriptableObject, but for example purposes I thought this is the simplest working case

**Review by oleks-k — APPROVED** [review]
Approving with the changes.

**Pull request merged**